enhancements/update/configurable-clusterversion-signature-stores.md

```diff
@@ -0,0 +1,194 @@
+
+### Non-Goals
+
+* This enhancement does not cover how OSUS running in the disconnected environment gets the signatures and if the 
+signatures are validated during the mirroring process.
+
+## Proposal
+
+### Workflow Description
+
+1. Cluster admin will have to add the verification URI to CVO
+2. CVO will reach the newly added URI to verify the release images.
+
+### API Extensions
+
+* CVO will grow a new field, an array, which will be populated with the image verification URI.
+* This will be an array as it'll allow the admin to use multiple locally configured stores. 
+* Existing default signature stores will be used as fallback in-case custom stores don't contain signatures or are offline. 
+* If this property is empty, default stores will be used, similar to how we handle upstream for update service today.
+* Admins will still be able to use the existing config maps for signature verification. 
+* CVO will check the release signatures in the local ConfigMap first and will reach in parallel to other signature stores.
+
+This enhancement proposes to modify the [api/config/v1/types_cluster_version.go](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go) as following,
+
+A new field `SignatureStores` will be added to the [`ClusterVersionSpec`](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go#L40-L96) structure:
+
+```go
+type ClusterVersionSpec struct {
+	...
+	// SignatureStores contains the upstream URIs to verify release signatures.
+	// By default, CVO will use existing signature stores if this property is empty.
+	// The CVO will check the release signatures in the local ConfigMaps first. It will search for a valid signature 
+	// in these stores in parallel only when local ConfigMaps did not include a valid signature.
+	// Validation will fail if none of the signature stores reply with valid signature before timeout.
+	//
+	// Items in this list should be a valid absolute http/https URI of an upstream signature store as per rfc1738.
+	// +kubebuilder:validation:Format=XValidation:rule="self.all(x, isURL(x))",message="signatureStores must contain only valid absolute URLs per the Go net/url standard"
+	// +listType=set
+	// +optional
+	SignatureStores [32]string `json:"signatureStores,omitempty"`
+}
+```
+
+
+### Risks and Mitigations
+
+* Users will have to make sure that they've cloned all the signatures needed for image verification or use some of the 
+existing stores along with custom stores as missing signatures will cause the verification to fail.
+* We recommend admin to use [oc-mirror] clone all releases and release signatures as well building the graph-data 
+container image with the signatures
+
+### Drawbacks
+
+N/A.
+
+## Design Details
+
+### Test Plan
+
+The CVO logic for image verification remains the same, it's just reaching a different signature store than preconfigured ones.
+Thus, explicit e2e tests are not needed for this enhancement.
```

Comment:
This doesn't sound like a particularly strong argument for me, will QE at least be testing this feature?

Reply:
Yes, QE will be testing this feature and will verify if the default stores are being replaced with new stores.

Reply:
You should make sure to include in this section that QE will be testing, and it would be helpful as well to consider the test cases that you might want them to run, that's what this section is for

Reply:
Added how QE will be checking this functionality